shell/src/main/scala/ammonite/shell/ShellSession.scala

```diff
@@ -5,27 +5,37 @@ import java.nio.file.attribute.PosixFilePermission
 
 import ammonite.ops._
 import ammonite.repl.FrontEndUtils
-import pprint.{Config, PPrinter, PPrint}
+import pprint.{Config, PPrinter}
+
+import scala.util.Try
 
 case class ShellSession() extends OpsAPI {
   var wd0 = pwd
   /**
-   * The current working directory of the shell, that will get picked up by
-   * any ammonite.ops commands you use
-   */
+    * The current working directory of the shell, that will get picked up by
+    * any ammonite.ops commands you use
+    */
   implicit def wd = wd0
   /**
-   * Change the working directory `wd`; if the provided path is relative it
-   * gets appended on to the current `wd`, if it's absolute it replaces.
-   */
+    * Change the working directory `wd`; if the provided path is relative it
+    * gets appended on to the current `wd`, if it's absolute it replaces.
+    */
   val cd = (arg: Path) => {
-    if (!stat(arg).isDir) throw new NotDirectoryException(arg.toString)
-    else {
-      wd0 = arg
-      wd0
-    }
+    /*
+     * `arg` should be a directory or a symlink to a directory
+     * `realPath will be None if that is not the case,
+     * otherwise - Some(realPath)
+     */
+    val realPath = Option(arg)
+      .filter(_.isDir)
+      .orElse(Try(Path(arg.toNIO.toRealPath())).toOption.filter(_.isDir))
 
+    realPath match {
+      case None => throw new NotDirectoryException(arg.toString)
+      case Some(path) => wd0 = path; wd0
```

Comment:
I think instead of `wd0 = path` this should be `wd0 = arg` because then you can actually change into a symlinked directory and not into the directory the symlink points to.

As it is now, `tmp` is symlinked on a Mac OS X, and you can't do something like this:
```
cd! root/'tmp
ls! up/home // "up" does not point to root
```